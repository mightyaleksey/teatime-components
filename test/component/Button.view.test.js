'use strict';

const Button = require('../../component/Button');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(Button.defaultProps.size).toBe('m');
});

test('defaultProp #theme should have a value `normal`', () => {
  expect(Button.defaultProps.theme).toBe('normal');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Button>
      Make Awesome!
    </Button>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Button
      autoFocus
      className='outer'
      disabled
      id='buttons-id'
      type='submit'>
      Make Awesome!
    </Button>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
