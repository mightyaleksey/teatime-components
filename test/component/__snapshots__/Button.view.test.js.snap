// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<button
  autoFocus={undefined}
  className="control"
  disabled={undefined}
  id={undefined}
  role="button"
  type="button"
>
  Make Awesome!
</button>
`;

exports[`Renders correctly with defined props 1`] = `
<button
  autoFocus={true}
  className="control outer"
  disabled={true}
  id="buttons-id"
  role="button"
  type="submit"
>
  Make Awesome!
</button>
`;
